src/test/java/seedu/command/FindCommandTest.java

```diff
@@ -0,0 +1,149 @@
+import seedu.ui.Ui;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.common.Constants.TAB;
+
+public class FindCommandTest {
+
+    TaskList tasks = new TaskList();
+    Ui ui = new Ui();
+    Storage storage = new Storage();
+
+    @Test
+    public void findTaskInTitle() {
+
+        String input = "add n/testForfInd t/11:15-13:15 d/2024-02-29";
+        Command command = new AddCommand(input);
+
+        command.setCommandVariables(tasks, storage, ui);
+        try {
+            command.execute();
+        } catch (ProjException e) {
+            assertTrue(false);
+        }
+
+        command = new FindCommand("find find");
+        command.setCommandVariables(tasks, storage, ui);
+        String output = "";
+        try {
+            output = command.execute().getFeedback();
+        } catch (ProjException e) {
+            assertTrue(false);
+        }
+        String expectedOutput = TAB + "Found the followings tasks:\n" + TAB + TAB +
+                "1. [TODO] Title: testForfInd | 2024-02-29 : 11:15 - 13:15\n";
+        assertEquals(expectedOutput,output);
+    }
+
+    @Test
+    public void findTaskInDescription() {
+
+        String input = "add n/test i/FinDDes t/11:15-13:15 d/2024-02-29";
+        Command command = new AddCommand(input);
+
+        command.setCommandVariables(tasks, storage, ui);
+        try {
+            command.execute();
+        } catch (ProjException e) {
+            assertTrue(false);
+        }
+
+        command = new FindCommand("find dEs");
+        command.setCommandVariables(tasks, storage, ui);
+        String output = "";
+        try {
+            output = command.execute().getFeedback();
+        } catch (ProjException e) {
+            assertTrue(false);
+        }
+        String expectedOutput = TAB + "Found the followings tasks:\n" + TAB + TAB +
+                "1. [TODO] Title: test | Description: FinDDes  | 2024-02-29 : 11:15 - 13:15\n";
+        assertEquals(expectedOutput,output);
+    }
+
+    @Test
+    public void findTaskInLocation() {
+
+        String input = "add n/test l/NUScom2";
+        Command command = new AddCommand(input);
+
+        command.setCommandVariables(tasks, storage, ui);
+        try {
+            command.execute();
+        } catch (ProjException e) {
+            assertTrue(false);
+        }
+
+        command = new FindCommand("find COM");
+        command.setCommandVariables(tasks, storage, ui);
+        String output = "";
+        try {
+            output = command.execute().getFeedback();
+        } catch (ProjException e) {
+            assertTrue(false);
+        }
+        String expectedOutput = TAB + "Found the followings tasks:\n" + TAB + TAB +
+                "1. [TODO] Title: test | Location: NUScom2 \n";
+        assertEquals(expectedOutput,output);
+    }
+
+    @Test
+    public void findTask_noPatternInput_outputError() {
```

Comment:
Is there no need for Camel-Case naming here?

Reply:
There are two type of error naming: 
1. Camel-Case
2. Java code normally use `camelCase` for method names e.g., testStringConversion but when writing test methods, sometimes another convention is used: `whatIsBeingTested_descriptionOfTestInputs_expectedOutcome `e.g., `intDivision_zeroDivisor_exceptionThrown`
![image](https://user-images.githubusercontent.com/49268567/78994224-1a8a9780-7b72-11ea-8e30-f9989caa315f.png)
Please check week 8 contents if there is anything that you may not understand.